docs/guide/testing.md

```diff
@@ -0,0 +1,168 @@
+# Testing
+
+Code testing is an important part of software development.
+This chapter will cover how to write tests for your Textual apps.
+
+## What is testing?
+
+It is common to write tests alongside your app.
+A *test* is simply a function that confirms your app is working correctly.
+
+!!! tip "Learn more about testing"
+
+    We recommend [Python Testing with pytest](https://pythontest.com/pytest-book/) for a comprehensive guide to writing tests.
+
+## Do you need to write tests?
+
+The short answer is "no", you don't *need* to write tests.
+
+In practice however, it is almost always a good idea to write tests.
```

Comment:
```suggestion
1. In practice however, it is almost always a good idea to write tests.
```

Reply:
Not keen on a list here. Wasn't intended to be list like.